### PR TITLE
Add invalidate name, tests currently failing

### PR DESCRIPTION
**Pull request by maurelian**

### CODE REVIEW

File: src/test/test.js [hunk 1]
```diff
@@ -411,9 +429,55 @@ describe('Registrar', () => {
...
+                params: [2 * 86400],  // 86400 seconds in a day
+                id: Date.now()
+              }, () => {
+                timeDiff += 8 * 7 * 86400;
+                registrar.getEntry('foo', (entryErr1, entryResult1) => {
+                  assert.equal(entryErr1, null);
+                  assert.equal(entryResult1.name, 'foo');
+                  assert.ok(entryResult1.deed.owner.slice(0,2) == '0x');
+                  assert.equal(entryResult1.mode, 'forbidden-can-invalidate');
+                  registrar.invalidateName('foo', { from: accounts[1], gas: 4700000 }, (invalidateErr, invalidateResult) => {
```
Comment (maurelian) [inline src/test/test.js]: @alexvandesande I've added the `registrar.invalidateName` method to the lib, but for whatever reason I'm unable to get this test to pass. It's returning an invalid opcode error. Appreciate if you can take a look.

**Comment by maurelian** [comment]
@alexvandesande @Arachnid would appreciate some eyes on this when you have a chance. 

As requiest I added an invalidateName function, which should be really simple since it only takes a single argument `unhashedName`, but I can't successfully call it on the registrar.

File: src/test/test.js [hunk 2]
```diff
@@ -464,12 +464,17 @@ describe('Registrar', () => {
...
                     assert.equal(entryResult1.name, 'foo');
                     assert.ok(entryResult1.deed.owner.slice(0,2) == '0x');
-                    assert.equal(entryResult1.mode, 'forbidden-can-invalidate');
+                    // the `forbidden-can-invalidate` mode does not exist in the registrar 
+                    // but is useful in this context where we know the plain text name
+                    assert.equal(entryResult1.mode, 'forbidden-can-invalidate'); 
                     debugger;
+                    // status `2` corresponds to 'owned'
                     assert.equal(entryResult1.status, 2);
                     registrar.invalidateName('foo', { from: accounts[1], gas: 4700000 }, (invalidateErr, invalidateResult) => {
```
Comment (maurelian) [inline src/test/test.js]: This call is failing, despite the checks in the previous steps to verify that is should be possible to invalidate.

Comment (Arachnid) [reply src/test/test.js]: You're passing the hash of the name to `invalidateName`, not the name itself. :)

**Pull request merged**